script.js

```diff
@@ -1,12 +1,304 @@
+function makePageForEpisodes(showId) {
+  rootElem.textContent = ""; // Set the of the root to blank as my the root content is the parent of my elements
+
+  if (showId != 0) {
+    /*This will calls the Async function gives it show id parameter*/
+    fetchEpisodesAsync(showId).then((episodesData) => {
+      // receives a episodesData
+
+      episodeDropList(episodesData, showId); //  calls episode drop list function Passes episodesData and showId
+      episodesData.forEach((e) => {
+        // loops through the episodes data
+
+        //calls the function that creates html for episodes
+        setEpisodesHTML(e);
+      });
+    });
+  } else {
+    makePageForShows(allShows); // displays all the shows.
+  }
+}
+
+/*This function will create the basic structure of one episode Title image and description that takes element as an argument*/
+
+/*4th creates episode content */
+function setEpisodesHTML(element) {
+  //console.log(element);
+  //Elements are the object inside of the array.
+  let theContainer = document.createElement("div"); //Creates an div element assign it assign it to the container element.
+  theContainer.setAttribute("value", "");
+  theContainer.setAttribute("class", "xl-col-4 placeHolder"); // Assign the Class of "xl-col-4" from the grid CSS file.
+  rootElem.appendChild(theContainer); // Appended the div element to the parent element of root.
+
+  let theArticle = document.createElement("article"); // Creates an article element.
+  theArticle.id = "article"; //Assigned the id of article to the article.
+  theArticle.setAttribute("class", "artContainer"); // Assigned the class of artContainer to the Article element.
+  theContainer.appendChild(theArticle); // Appended the Article to the div of our placeholder.
+
+  let theTitleTagName = document.createElement("h3"); //Created the a h3 element to hold the title.
+  theArticle.appendChild(theTitleTagName); // Appended the h3 element to the article Container.
+  theTitleTagName.textContent = `${element.name}-${titleMaker(
+    element.season,
+    element.number
+  )}`; // Set the the text content of the h3 element the object name property called the title maker maker function
+  theTitleTagName.id = "title"; // Set the id of h3 element to 'title'.
+
+  let theImageTagName = document.createElement("img"); // created an image element.
+  theArticle.appendChild(theImageTagName); // Appended the image tag to the article parents.
+  theImageTagName.id = "image"; //Set an id of 'image to it'
+
+  var noImage = "";
+  element.image == null ? noImage : (noImage = element.image.medium);
+  theImageTagName.src = noImage;
+  //Assigned the medium image object property to img element.
+  let theParagraph = document.createElement("p"); // Created Paragraph element.
+
+  theParagraph.textContent = element.summary.replace(/(<([^>]+)>)/gi, ""); //set the paragraph text content to the summary object property and remove the preset p tag using replace method
```

Comment:
I like the way you put comment for every thing. I usually write much less comments.

Reply:
There is a [big debate](https://stackoverflow.com/questions/209015/what-is-self-documenting-code-and-can-it-replace-well-documented-code) about how many comments are the right amount. Some people argue for many comments, some even argue for no comments ever! I think it is a personal preference issue. At this stage in your learning, I would advise that if they are helpful for you then please use them as much as you need.